https://bugs.gentoo.org/962751
https://github.com/xbmc/xbmc/pull/27224

Modified to apply to 22.0_alpha1 which doesn't include.
https://github.com/xbmc/xbmc/commit/8a27124db43ec10415b21aa6c7a15b0b6aded656

From 8b44b36f7bfe6e32770c7df5f956a1da9fcaaf61 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Wed, 10 Sep 2025 19:16:17 +0200
Subject: [PATCH] [filesystem] Switch to long for CURLPROXY_* enums

curl 8.16.0 bumped the enums to long, see: https://github.com/curl/curl/pull/18054

This fixes the build with curl 8.16+ for the following failure:

	/builds/alpine/aports/community/kodi/src/xbmc-21.2-Omega/xbmc/filesystem/CurlFile.cpp:48:5: error: cannot initialize an array element of type 'curl_proxytype' with an rvalue of type 'long'
	   48 |     CURLPROXY_HTTP,   CURLPROXY_SOCKS4,          CURLPROXY_SOCKS4A,
	      |     ^~~~~~~~~~~~~~
	/usr/include/curl/curl.h:791:35: note: expanded from macro 'CURLPROXY_HTTP'
	  791 | #define CURLPROXY_HTTP            0L /* added in 7.10, new in 7.19.4 default is
	      |                                   ^~
	...
--- a/xbmc/filesystem/CurlFile.cpp
+++ b/xbmc/filesystem/CurlFile.cpp
@@ -46,7 +46,7 @@ using namespace std::chrono_literals;
 #define FITS_INT(a) (((a) <= INT_MAX) && ((a) >= INT_MIN))

-static const auto proxyType2CUrlProxyType = std::unordered_map<XFILE::CCurlFile::ProxyType, int>{
+static const auto proxyType2CUrlProxyType = std::unordered_map<XFILE::CCurlFile::ProxyType, long>{
     {CCurlFile::ProxyType::HTTP, CURLPROXY_HTTP},
     {CCurlFile::ProxyType::SOCKS4, CURLPROXY_SOCKS4},
     {CCurlFile::ProxyType::SOCKS4A, CURLPROXY_SOCKS4A},
